pkg/route/api/validation/validation.go

```diff
@@ -11,9 +13,19 @@ import (
 	"k8s.io/kubernetes/pkg/util/validation/field"
 
 	oapi "github.com/openshift/origin/pkg/api"
+	"github.com/openshift/origin/pkg/cmd/util"
 	routeapi "github.com/openshift/origin/pkg/route/api"
 )
 
+// disableRouteTLSValidation returns whether or not to disable TLS config validation checks.
+func disableRouteTLSValidation() bool {
```

Comment:
Why not just create a new function and invoke it from the router in the unique_host check (or in an optional pre check step, based on a flag)?